src/provider_completions.jl

```diff
@@ -1,38 +1,56 @@
 function process(r::JSONRPC.Request{Val{Symbol("textDocument/completion")},TextDocumentPositionParams}, server)
     tdpp = r.params
     line = get_line(tdpp, server)
-    if line==""
-        completion_list = CompletionItem[]
-    else
-        comp = Base.REPLCompletions.completions(line, chr2ind(line,tdpp.position.character))[1]
-        n = length(comp)
-        comp = comp[1:min(length(comp),25)]
-        CIs = map(comp) do i
-            s = get_sym(i)
-            d = ""
-            d = get_docs(s)
-            d = isa(d,Vector{MarkedString}) ? (x->x.value).(d) : d
-            d = join(d[2:end],'\n')
-            d = replace(d,'`',"")
-        
-            kind = 6
-            if isa(s, String)
-                kind = 1
-            elseif isa(s, Function)
-                kind = 3
-            elseif isa(s, DataType)
-                kind = 7
-            elseif isa(s, Module)
-                kind = 9
-            elseif isa(s, Number)
-                kind = 12
-            elseif isa(s, Enum)
-                kind = 13
-            end
-            CompletionItem(i, kind, d)
+    
+    word = IOBuffer()
+    for c in reverse(line[1:chr2ind(line,tdpp.position.character)])
+        if c=='\\'
+            write(word, c)
+            break
         end
+        if !(Base.is_id_char(c) || c=='.' || c=='_' || c=='^')
+            break
+        end
+        write(word, c)
+    end
+    
+    str = reverse(takebuf_string(word))
+    prefix = str[1:findlast(word,'.')]
+    comp = Base.REPLCompletions.completions(str,length(str.data))[1]
```

Comment:
Maybe use ``endof(str)`` instead of ``length`` here? Probably safer to not assume a certain storage format here, I wouldn't be surprised if they changed the internals of ``Strings`` again at some future point.